### ISSUE TITLE
Damage Source Crash

### ISSUE THREAD

**Issue by WenXin20** [XiaoHuNao/EnemyBanner#3]
```
Minecraft v1.20.1
Forge v47.2.20
Enemy Banner v0.0.3
```

After loading my world, shortly afterwards the game crashed from a null damage source. I am not sure what triggered it.
[Crash Report](https://gist.github.com/WenXin20/4c3162f66167f0f35add43e4fdfc5622)

**Comment by WenXin20** [comment]
This also seems to trigger when breaking blocks
```
[22Apr2024 00:43:26.373] [Server thread/ERROR] [net.minecraft.network.protocol.PacketUtils/]: Failed to handle packet net.minecraft.network.protocol.game.ServerboundPlayerActionPacket@7b292371, suppressing error
java.lang.NullPointerException: Cannot invoke "net.minecraft.world.damagesource.DamageSource.m_7639_()" because "damageSource" is null
	at com.xiaohunao.enemybanner.EnemyBannerEventSubscriber.looting(EnemyBannerEventSubscriber.java:117) ~[enemybanner-1.20.1-0.0.2.jar%23639!/:1.20.1-0.0.2]
	at com.xiaohunao.enemybanner.__EnemyBannerEventSubscriber_looting_LootingLevelEvent.invoke(.dynamic) ~[enemybanner-1.20.1-0.0.2.jar%23639!/:1.20.1-0.0.2]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.common.ForgeHooks.getLootingLevel(ForgeHooks.java:331) ~[forge-1.20.1-47.2.20-universal.jar%23909!/:?]
	at net.minecraftforge.common.ForgeHooks.getLootingLevel(ForgeHooks.java:324) ~[forge-1.20.1-47.2.20-universal.jar%23909!/:?]
	at net.minecraft.world.level.storage.loot.LootContext.getLootingModifier(LootContext.java:93) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceWithLootingCondition.test(LootItemRandomChanceWithLootingCondition.java:34) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceWithLootingCondition.test(LootItemRandomChanceWithLootingCondition.java:16) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at java.util.function.Predicate.lambda$and$0(Predicate.java:69) ~[?:?]
	at net.minecraftforge.common.loot.LootModifier.apply(LootModifier.java:58) ~[forge-1.20.1-47.2.20-universal.jar%23909!/:?]
	at net.minecraftforge.common.ForgeHooks.modifyLoot(ForgeHooks.java:1186) ~[forge-1.20.1-47.2.20-universal.jar%23909!/:?]
	at net.minecraft.world.level.storage.loot.LootTable.m_230922_(LootTable.java:118) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.storage.loot.LootTable.m_287195_(LootTable.java:112) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.block.state.BlockBehaviour.m_49635_(BlockBehaviour.java:258) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.m_287290_(BlockBehaviour.java:774) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.block.Block.m_49874_(Block.java:258) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.block.Block.dropResources(Block.java:286) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.world.level.block.Block.m_6240_(Block.java:356) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.level.ServerPlayerGameMode.m_9280_(ServerPlayerGameMode.java:256) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.level.ServerPlayerGameMode.m_215116_(ServerPlayerGameMode.java:218) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.level.ServerPlayerGameMode.m_214168_(ServerPlayerGameMode.java:188) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.m_7502_(ServerGamePacketListenerImpl.java:1020) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.network.protocol.game.ServerboundPlayerActionPacket.m_5797_(ServerboundPlayerActionPacket.java:42) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.network.protocol.game.ServerboundPlayerActionPacket.m_5797_(ServerboundPlayerActionPacket.java:8) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.network.protocol.PacketUtils.m_263899_(PacketUtils.java:22) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:770) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:161) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:753) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:747) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:115) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:732) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:665) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[client-1.20.1-20230612.114412-srg.jar%23904!/:?]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
	```

**Comment by XiaoHuNao** [comment]
I have tried to fix, please update 0.0.4 version

**Comment by WenXin20** [comment]
This is fixed, thanks!
